src/vcpkg/build.cpp

```diff
@@ -951,10 +951,11 @@ namespace vcpkg::Build
                                                                             action.public_abi(),
                                                                             std::move(find_itr->second));
 
-        if (error_count != 0)
+        if (error_count != 0 && action.build_options.backcompat_features == BackcompatFeatures::PROHIBIT)
```

Comment:
Maybe a little bit too lax?

Reply:
The thing is, for any future lint check we can't make them mandatory because a user could have an older baseline set -- the exact problem indicated by @horenmar -- and I don't see a reason this check is particularly more or less severe than any of the other lint checks here.

Reply:
Yeah I thought of that all new checks are only active when `backcompat_features == BackcompatFeatures::PROHIBIT`but that old checks are not effected

Reply:
I can do that ... but then we have to remember to update this whenever we change the effects of any of those checks, etc. Let me confirm with the other maintainers if this is OK.